### PR TITLE
Making the release process more automated

### PR DESCRIPTION
**Pull request by ppathan**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
* With this change cli will replace the git tag and docker tag at runtime based on the current release version
* This change needs to be matched with aureiotsuite as well

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/k8smanager.ts [hunk 1]
```diff
@@ -224,10 +224,14 @@ export class K8sManager implements IK8sManager {
                     promises.push(this._api.createNamespacedReplicationController(this._namespace, doc));
                     break;
                 case 'Deployment':
-                    const imageName: string = doc.spec.template.spec.containers[0].image;
+                    let imageName: string = doc.spec.template.spec.containers[0].image;
                     if (imageName.includes('{runtime}')) {
                         doc.spec.template.spec.containers[0].image = imageName.replace('{runtime}', this._config.Runtime);
                     }
+                    imageName = doc.spec.template.spec.containers[0].image;
```
Comment (avanikp) [inline src/k8smanager.ts]: This is the same as in line 227 -
imageName = doc.spec.template.spec.containers[0].image;

remove ?

File: src/index.ts [hunk 2]
```diff
@@ -225,24 +227,36 @@ function main() {
                     }
                 })
                 .then(({appId, domainName, objectId, servicePrincipalId, servicePrincipalSecret}) => {
+                    cachedAuthResponse.options.tokenAudience = null;
+                    answers.deploymentSku = program.sku;
+                    answers.runtime = program.runtime;
+                    if (program.versionOverride === 'master') {
```
Comment (avanikp) [inline src/index.ts]: other than 'master' what are the other valid input values that are expected ? Should we make this an enum instead ?

File: solutions/remotemonitoring/armtemplates/basic-static-map.json [hunk 3]
```diff
@@ -207,9 +207,23 @@
...
+            "defaultValue": "master",
+            "metadata": {
+                "description": "The release version is used for repoURL for reverse-proxy-dotnet, vmScriptUri and the docker image version"
+            }
+        },
+        "pcsDockerTag": {
+            "type": "string",
+            "defaultValue": "testing",
+            "metadata": {
+                "description": "The release version is used for repoURL for reverse-proxy-dotnet, vmScriptUri and the docker image version"
```
Comment (avanikp) [inline solutions/remotemonitoring/armtemplates/basic-static-map.json]: update description

File: solutions/remotemonitoring/armtemplates/basic.json [hunk 4]
```diff
@@ -207,9 +207,23 @@
...
+            "defaultValue": "master",
+            "metadata": {
+                "description": "The release version is used for repoURL for reverse-proxy-dotnet, vmScriptUri and the docker image version"
+            }
+        },
+        "pcsDockerTag": {
+            "type": "string",
+            "defaultValue": "testing",
+            "metadata": {
+                "description": "The release version is used for repoURL for reverse-proxy-dotnet, vmScriptUri and the docker image version"
```
Comment (avanikp) [inline solutions/remotemonitoring/armtemplates/basic.json]: same, update description text

File: src/k8smanager.ts (hunk 1 above)
Comment (ppathan) [reply src/k8smanager.ts]: Yes it is expected. Basically i need to reassign with the changed value online 229

File: src/index.ts (hunk 2 above)
Comment (ppathan) [reply src/index.ts]: I actually only want master branch to be the valid value if it is not using a published version.

**Pull request merged**